src/fidesops/ops/service/saas_request/override_implementations/firebase_auth_request_overrides.py

```diff
@@ -0,0 +1,155 @@
+def firebase_auth_user_access(
+    node: TraversalNode,
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    input_data: Dict[str, List[Any]],
+    secrets: Dict[str, Any],
+) -> List[Row]:
+    """
+    SaaS Request Override function used to integrate access requests with
+    Firebase Auth service. Specifically, it reads `User` records.
+    """
+    app = initialize_firebase(secrets)
+
+    processed_data = []
+
+    emails = input_data.get("email", [])
+    for email in emails:
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        row = {"email": user.email, "uid": user.uid}
+        if user.display_name is not None:
+            row["display_name"] = user.display_name
+        if user.phone_number is not None:
+            row["phone_number"] = user.phone_number
+        if user.disabled is not None:
+            row["disabled"] = user.disabled
+        if user.email_verified is not None:
+            row["email_verified"] = user.email_verified
+        if user.provider_data:
+            pds = []
+            for pd in user.provider_data:
+                pds.append(
+                    {
+                        "display_name": pd.display_name,
+                        "provider_id": pd.provider_id,
+                        "email": pd.email,
+                    }
+                )
+            row["provider_data"] = pds
+        processed_data.append(row)
+
+    return processed_data
+
+
+@register("firebase_auth_user_update", [SaaSRequestType.UPDATE])
+def firebase_auth_user_update(
+    param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    secrets: Dict[str, Any],
+) -> int:
+    """
+    SaaS Request Override function used to integrate update requests with
+    Firebase Auth service. Specifically, it updates `User` records.
+    """
+    app = initialize_firebase(secrets)
+    rows_updated = 0
+    # each update_params dict correspond to a record that needs to be updated
+    for row_param_values in param_values_per_row:
+        email = row_param_values.get("email")
+        user: UserRecord = auth.get_user_by_email(email, app=app)
+        masked_fields = row_param_values["masked_object_fields"]
+        display_name = masked_fields.get("display_name", user.display_name)
+        phone_number = masked_fields.get("phone_number", user.phone_number)
+        disabled = masked_fields.get("disabled", user.disabled)
+        email_verified = masked_fields.get("email_verified", user.email_verified)
+
+        auth.update_user(
+            user.uid,
+            email=email,
+            display_name=display_name,
+            phone_number=phone_number,
+            disabled=disabled,
+            email_verified=email_verified,
+            app=app,
+        )
+        rows_updated += 1
+    return rows_updated
+
+
+@register("firebase_auth_user_delete", [SaaSRequestType.DELETE])
+def firebase_auth_user_delete(
```

Comment:
as noted above, this isn't actually being leveraged currently. but i figured i'd keep the function implementation here just for reference, in case we do want to go down this route at some point. happy to include this in some other manner, but didn't want this work to get lost.

Reply:
yup, that's how i felt about it too. maybe it's worth me throwing in a comment here just to clarify it moving forward.